libraries/lib-wave-track/WaveTrack.cpp

```diff
@@ -318,6 +460,18 @@ bool WaveTrack::Interval::IsPlaceholder() const
    return mpClip->GetIsPlaceholder();
 }
 
+const Envelope& WaveTrack::Interval::GetEnvelope() const
+{
+   return *mpClip->GetEnvelope();
+}
+
+void WaveTrack::Interval::SetEnvelope(const Envelope& envelope)
+{
+   ForEachClip([&](auto& clip) {
+      clip.SetEnvelope(std::make_unique<Envelope>(envelope));
+   });
```

Comment:
Actually this is needed only for clip 0.  See the commits merged at 48154efe6d8ce8e02c570a665a31caab39786cdc.  Right channel envelopes are just ignored.